core/init.js

```
@@ -30,8 +31,9 @@ $.init = function() {
   $.ctx = $.canvas.getContext('2d');
 
   $.groups = {};
-  $.input = new Input();
-  $.cam = new Camera($.vw, $.vh);
+  let targetElement = D.body;
+  $.input = Input.build(targetElement);
+  $.cam = new Camera($, $.vw, $.vh, 1024, 1024);
```

Maybe we should remove the world size from the constructor of the camera and let the users set it out later in the game (because not all worlds have the same size). To ensure they set those values, we can throw an exception if the camera is being used without that. What do you think?

---

I think k3b.js where I simple rewrote init.js doesn't have a clearly defined role - it's just something that holds objects when no other place is instantly obvious. I didn't propose any role for k3j because I don't know your intent well enough and hence I'm postponing as many design changes as I can.

---

Exception - I'm trying to code with a mindset that  we're the only consumers of this framework so I'm fine with not being defensive in code at all.

---

I think that the role of init.js (or k3j.js) is basically to initialize all the components that don't require user input or game-dependant variables. For example, all games need a camera, and input, right? So, we can initialize those modules right away and call methods later to set the viewport size or bind the keys.

Does it make sense?